sweep/aggregator.go

```diff
@@ -540,19 +547,21 @@ func (b *BudgetAggregator) ClusterInputs(inputs InputsMap) []InputSet {
 	// NOTE: cannot pre-allocate the slice since we don't know the number
 	// of input sets in advance.
 	inputSets := make([]InputSet, 0)
-	for _, cluster := range clusters {
+	for height, cluster := range clusters {
 		// Sort the inputs by their economical value.
 		sortedInputs := b.sortInputs(cluster)
 
 		// Create input sets from the cluster.
-		sets := b.createInputSets(sortedInputs)
+		sets := b.createInputSets(sortedInputs, height)
 		inputSets = append(inputSets, sets...)
 	}
 
 	// Create input sets from the exclusive inputs.
-	for _, input := range exclusiveInputs {
-		sets := b.createInputSets([]SweeperInput{*input})
-		inputSets = append(inputSets, sets...)
+	for _, cluster := range exclusiveInputs {
```

Comment:
I think the way this is written, heights won't get batched together? Each cluster from `exclusiveInputs` will only have a single entry